modules/social_features/social_group/src/CurrentGroupService.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\social_group;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\Context\ContextProviderInterface;
+use Drupal\group\Entity\GroupInterface;
+
+/**
+ * Centralized ways to get the current group.
+ *
+ * @package Drupal\social_group.
+ */
+class CurrentGroupService {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  private EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * The group route context.
+   *
+   * @var \Drupal\Core\Plugin\Context\ContextProviderInterface
+   */
+  private ContextProviderInterface $groupRouteContext;
+
+  /**
+   * Constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Plugin\Context\ContextProviderInterface $group_route_context
+   *   The group route context.
+   */
+  public function __construct(
+        EntityTypeManagerInterface $entity_type_manager,
+        ContextProviderInterface $group_route_context,
+    ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->groupRouteContext = $group_route_context;
+  }
+
+  /**
+   * Get group from runtime contexts.
+   *
+   * @return \Drupal\group\Entity\GroupInterface|null
+   *   The current group or NULL.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function fromRunTimeContexts(): ?GroupInterface {
+    $runtime_context = $this->groupRouteContext->getRuntimeContexts([]);
+    if (isset($runtime_context['group']) === FALSE) {
+      return NULL;
+    }
+
+    $group = $runtime_context['group']->getContextData()->getValue();
+    if ($group instanceof GroupInterface) {
+      return $group;
+    }
+
+    if (is_int($group) === TRUE) {
```

Comment:
There's one stylistic thing I disagree with and one functional issue in the code:

Stylistically: I'm unsure why the `=== TRUE` is added here since that's the default for if-statements. Especially since `is_int` returns an exact boolean and no truthy values. `if (is_int($group))` is less complex to view and equivalent so it should be preferred.

Functionally: This code will never be true. The group route service resolves the context and returns `\Drupal\group\Entity\GroupInterface|null` as type for the context service.

In the old code we would get the parameter from the route directly. That can be an integer in the case of views (which doesn't configure upcasting for its arguments). However, the group implementation converts that to `NULL` due to the following line of code in `GroupRouteContextTrait::getGroupFromRoute`: `if (($group = $route_match->getParameter('group')) && $group instanceof GroupInterface) {`.

Fixing that on our side is slightly more difficult because it means that things that are used in views (Actions/Blocks/Forms) should use the plugin context system or the form context system rather than getting the context from the route. That way they get the instance of the context that Views itself has converted into an entity. So basically anything that might need this service for this code-path, shouldn't use this service.